src/test/scala/org/constellation/infrastructure/endpoints/TransactionEndpointsTest.scala

```diff
@@ -0,0 +1,54 @@
+package org.constellation.infrastructure.endpoints
+
+import cats.effect.{ContextShift, IO}
+import io.circe.generic.auto._
+import org.http4s.circe.CirceEntityDecoder._
+import org.http4s.implicits._
+import org.constellation.ConstellationExecutionContext
+import org.constellation.domain.transaction.{LastTransactionRef, TransactionChainService, TransactionService}
+import org.constellation.infrastructure.p2p.client.TransactionClientInterpreter
+import org.constellation.util.Metrics
+import org.http4s.{HttpRoutes, Method, Request, Status, Uri}
+import org.mockito.{ArgumentMatchersSugar, IdiomaticMockito}
+import org.mockito.cats.IdiomaticMockitoCats
+import org.scalatest.{BeforeAndAfter, FreeSpec, Matchers}
+
+class TransactionEndpointsTest
+  extends FreeSpec
+    with IdiomaticMockito
+    with IdiomaticMockitoCats
+    with Matchers
+    with ArgumentMatchersSugar
+    with BeforeAndAfter {
+
+
+  implicit val cs: ContextShift[IO] = IO.contextShift(ConstellationExecutionContext.unbounded)
+
+  var transactionChainService: TransactionChainService[IO] = _
+  var transactionService: TransactionService[IO] = _
+  var metrics: Metrics = _
+  var transactionPeerEndpoints: HttpRoutes[IO] = _
+
+  before {
+    transactionChainService = mock[TransactionChainService[IO]]
+    transactionService = mock[TransactionService[IO]]
+    metrics = mock[Metrics]
+
+    transactionPeerEndpoints = TransactionEndpoints.peerEndpoints[IO](transactionService, metrics)
+  }
+
+  "transactionPeerEndpoints" - {
+    "getLastTransactionEndpoint" - {
+      "should return last transaction for a given address" in {
+        val lastTx = LastTransactionRef("abc", 1)
+        transactionService.transactionChainService shouldReturn transactionChainService
+        transactionChainService.getLastTransactionRef(*) shouldReturnF lastTx
+
+        val response = transactionPeerEndpoints.orNotFound.run(Request(Method.GET, uri"last-transaction/DAG123")).unsafeRunSync
+
+        response.status shouldBe Status.Ok
+        response.as[LastTransactionRef].unsafeRunSync shouldEqual lastTx
+      }
```

Comment:
endpoint shouldn't test internal stuff from the service but if the service has been correctly called. there should be another test testing the public interface of the service. thus it's called unit test :)
now this test test both endpoint and internal logic of transaction chain service

Reply:
Actually my goal was to check the endpoint correctness. I just needed it to return something.

Reply:
So to check the endpoint correctness, you should check if endpoint calls the proper service and that's the endpoint check. Then you should have separate test which checks if the service under this method returns correct data.